addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,16 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
```

Comment:
Also it's important that you keep the grouping by user, to send only one bus message with all the notifications for one user grouped together.